08.ls_object/lib/ls/formatter.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+require_relative 'args'
+require_relative 'file'
+
+module Ls
+  class Formatter
+    COLUMN_COUNT = 3
+
+    FILETYPES = {
+      'fifo' => 'p',
+      'characterSpecial' => 'c',
+      'directory' => 'd',
+      'blockSpecial' => 'b',
+      'file' => '-',
+      'link' => 'l',
+      'socket' => 's'
+    }.freeze
+
+    PERMISSIONS = {
+      '0' => '---',
+      '1' => '--x',
+      '2' => '-w-',
+      '3' => '-wx',
+      '4' => 'r--',
+      '5' => 'r-x',
+      '6' => 'rw-',
+      '7' => 'rwx'
+    }.freeze
+
+    private attr_reader :args # rubocop:disable Style/AccessModifierDeclarations
+
+    def initialize
+      @args = Ls::Args.new
+    end
+
+    def format
+      files = collect_files
+      args.long? ? format_long(files) : format_short(files)
+    end
+
+    private
+
+    def collect_files
+      pattern = args.pathname
+      params = args.all? ? [pattern, ::File::FNM_DOTMATCH] : [pattern]
+      file_paths = Dir.glob(*params).sort
+      args.reverse? ? file_paths.reverse! : file_paths
+      file_paths.map { |path| Ls::File.new(path) }
+    end
+
+    def format_long(files)
+      total_blocks = "total #{files.map(&:blocks).sum}"
+      max_lengths = %i[nlink user group size].to_h { |key| [key, find_max_length(files, key)] }
+      rows = collect_file_data(files).map { |data| format_row(data, max_lengths) }
+      [total_blocks, rows].join("\n")
+    end
+
+    def format_short(files)
+      max_basename_length = find_max_length(files, :basename)
+      file_names = files.map { |file| file.basename.ljust(max_basename_length) }
+
+      remainder = files.length % COLUMN_COUNT
+      (COLUMN_COUNT - remainder).times { file_names << [''] } unless remainder.zero?
+
+      row_count = file_names.length / COLUMN_COUNT
+      file_names.each_slice(row_count).to_a.transpose.map { |row| row.join("\t") }
+    end
+
+    def find_max_length(files, column)
+      collect_file_data(files).map { |data| data[column] }
+                              .max_by(&:length)
+                              .length
+    end
+
+    def collect_file_data(files)
+      files.map { |file| build_data(file) }
```

Comment:
```suggestion
      files.map { |file| sleep 1; build_data(file) }
```

ってやると、また改善ポイントが見つかるかも？